### PR TITLE
Add Dominik solutions

### PR DESCRIPTION
**Pull request by dbisiakowski**

### CODE REVIEW

File: dominik.clj [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+(defn find-the-odd-numbers-25 []
+  (= (#(filter odd? %) #{1 2 3 4 5}) '(1 3 5)))
+
+(defn reverse-a-sequence-23 []
+  (= (#(into () %) [1 2 3 4 5]) [5 4 3 2 1]))
+
+(defn palindrome-detector-27 []
+  (true? (#(= (apply str (reverse %)) (apply str %)) "racecar")))
+
+(defn fibonacci-sequence-26 []
```
Comment (IwoHerka) [inline dominik.clj:129]: Suggested formatting:
```
(defn fibonacci-sequence-26 []
  (let [fib (fn [arr num]
              (if (> num 2)
                (fib (conj arr (+ (first arr) (first (next arr)))) (dec num))
                (reverse arr)))]
    (= (fib '(1 1) 6) '(1 1 2 3 5 8))))
```

File: dominik.clj [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+(defn get-the-caps-29 []
+  (= ((fn [x] (clojure.string/replace x #"[^A-Z]" "")) "HeLlO, WoRlD!") "HLOWRD"))
+
+(defn duplicate-a-sequence-32 []
+  (= ((fn [x] (reduce #(conj (conj %1 %2) %2) [] x)) [[1 2] [3 4]]) '([1 2] [1 2] [3 4] [3 4])))
+
+(defn intro-to-some-48 []
+  (= 6 (some #{2 7 6} [5 6 7 8])))
+
+(defn intro-to-some-48 []
```
Comment (IwoHerka) [inline dominik.clj:152]: Reformat this anonymous function to `let` block.

File: dominik.clj [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        (if (> (count y) 0)
+          (if (= (first y) (second y))
+            (z x (drop 1 y))
+            (z (conj x (first y)) (drop 1 y))
+            )
+          (reverse x)
+          )
+        ) nil [1 1 2 3 3 2 2 3]) '(1 2 3 2 3)))
+
+(defn implement-range-34 []
```
Comment (IwoHerka) [inline dominik.clj:163]: Reformat this anonymous function to `let` block.

File: dominik.clj [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  (= ((fn ran [z x y]
+        (if (< x y)
+          (ran (conj z x) (inc x) y)
+          (reverse z)
+          )
+        ) '() -2 2) '(-2 -1 0 1)))
+
+(defn factorial-fun-42 []
```
Comment (IwoHerka) [inline dominik.clj:171]: Reformat this anonymous function to `let` block.

File: dominik.clj [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  (= ((fn [x] (partition-by identity x)) [:a :a :b :b :c]) '((:a :a) (:b :b) (:c))))
+
+(defn split-a-sequence-49 []
+  (= ((fn [numb x] [(into [] (take numb x)) (into [] (drop numb x))]) 1 [:a :b :c :d]) [[:a] [:b :c :d]]))
+
+(defn advanced-destructuring-31 []
+  (= [1 2 [3 4 5] [1 2 3 4 5]] (let [[a b & c :as d] [1 2 3 4 5]] [a b c d])))
+
+(defn a-half-truth-83 []
+  (= false ((fn [& args] (if (some true? args)
```
Comment (IwoHerka) [inline dominik.clj:213]: Reformat this anonymous function to `let` block.

File: dominik.clj [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+
+(defn rearranging-code-72 []
+  (= (apply + (map inc (take 3 (drop 2 [2 5 4 1 3 6]))))
+     (->> [2 5 4 1 3 6] (drop 2) (take 3) (map inc) (apply +))
+     11))
+
+(defn a-nil-key-134 []
+  (true? (#(nil? (get %2 %1 -1)) :a {:a nil :b 2})))
+
+(defn for-the-win-145 []
```
[the remark this answers is not in the document]
Comment (dbisiakowski) [reply dominik.clj:91]: Done

File: dominik.clj (hunk 1 above)
Comment (dbisiakowski) [reply dominik.clj:129]: Done

File: dominik.clj (hunk 2 above)
Comment (dbisiakowski) [reply dominik.clj:152]: Done

File: dominik.clj (hunk 3 above)
Comment (dbisiakowski) [reply dominik.clj:163]: Done

File: dominik.clj (hunk 4 above)
Comment (dbisiakowski) [reply dominik.clj:171]: Done

File: dominik.clj (hunk 5 above)
Comment (dbisiakowski) [reply dominik.clj]: Done

**Pull request merged**